### ISSUE TITLE
React Class Component example?

### ISSUE THREAD

**Issue by NotAmaan** [frontarm/navi#39]
Great library! I actually prefer having all the pages is JSON-ish format, plus you get the benefits of react-loadable built in which is perfect.

I was trying this out and wanted to see if there's an example of a route which uses a react class component instead of a function.

So instead of 
```
// pages/reference.js
import * as React from 'react'

export default function Reference() {
  return (
    <div>
      <h2>Reference</h2>
    </div>
  )
}
```

What would be the `createPage` syntax for this:
```
import React from 'react'

export default class Reference extends React.Component {
  render() {
    return <h2>Smart Page!</h2>;
  }
}
```

Thank you!

**Comment by jamesknelson** [comment]
I've just made a change to allow this. Previously, if `react-navi` saw a function as content, it would manually call it. I've kept this behavior for plain function, but now it'll check to see if the content is a class component and create an element if it is, passing through the `Route` object as props.

The new behavior is available as of react-navi 0.9. You can see a demo here:

https://frontarm.com/demoboard/?id=ec703f64-fa4f-4110-9323-8c46de76186f

**Comment by kmcq** [comment]
hi @jamesknelson, first this library is so awesome. We have a good-sized CRA web app that we now want to add some pages to that need SEO and this solution is oh so perfect for us.

my question! I'm wondering what you think about passing the `Route` object as props to pages that are function components too. We're converting our app from react-router to navi and have a lot of page components that use things like `this.props.match` from react-router to get a route param. It's a fine concession for our page components to be classes (and most of them already are!) but with the release of hooks we're excited to use fewer classes in the future.

Anyway, just wondering what you think!

Cheers!

**Comment by jamesknelson** [comment]
Hi @kmcq, glad you're finding some use out this :-)

I think you're right about treating content functions as components, especially now that hooks are about to be released. It'll be confusing if it's not possible to use hooks in content functions.

The issue I see here is that if a new content function is returned on each update by `getContent()` or a getter function for a switch, then it'll be treated as a new component by React for reconciliation purposes, which is going to break hooks anyway.

It might be that we should treat functions as components anyway, and just add something to the docs warning about the reconciliation issue.

I'll see if I can have a go fixing this over the next couple days, as I'd like Navi to be pretty much ready to go with hooks when they're released.

**Comment by jamesknelson** [comment]
Have just merged this change, so that `content` properties that are functions will now be treated as function components, and have a `route` property passed in.

This will be part of 0.11, which should be released soon after React 16.8 is (with hooks support).

**Comment by kmcq** [comment]
Yesss thank you! Getting the props passed in is sweet and those hooks you added are so awesome! 🔥 🎣 

Thanks for making these changes!
